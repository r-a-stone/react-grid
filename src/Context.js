import React, { createContext, useState } from "react";
import PropTypes from "prop-types";

const Context = createContext();

const propTypes = {
  columns: PropTypes.number,
  gutters: PropTypes.number,
  breakpoints: PropTypes.array
};

const defaultProps = {
  columns: 12,
  gutters: 24,
  breakpoints: [
    {
      name: "xs",
      minWidth: 0
    },
    {
      name: "sm",
      minWidth: 480
    },
    {
      name: "md",
      minWidth: 768
    },
    {
      name: "lg",
      minWidth: 1024
    }
  ]
};

const GridProvider = ({ children, ...props }) => {
  const [columns, setColumns] = useState(props.columns);
  const [gutters, setGutters] = useState(props.gutters);
  const [breakpoints, setBreakpoints] = useState(props.breakpoints);

  return (
    <Context.Provider
      value={{
        columns,
        gutters,
        breakpoints,

        setColumns,
        setGutters,
        setBreakpoints
      }}
    >
      {children}
    </Context.Provider>
  );
};

GridProvider.propTypes = propTypes;
GridProvider.defaultProps = defaultProps;

export default {
  Provider: GridProvider,
  Consumer: Context.Consumer
};
