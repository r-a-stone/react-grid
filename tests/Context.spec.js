import React from "react";
import renderer from "react-test-renderer";
import Context from "../src/Context";

it("Context renders without crashing.", () => {
  const component = renderer.create(
    <Context.Provider>
      <Context.Consumer>
        {context => {
          JSON.stringify(context);
        }}
      </Context.Consumer>
    </Context.Provider>
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
