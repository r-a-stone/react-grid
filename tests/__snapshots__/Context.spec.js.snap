// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Context renders without crashing. 1`] = `null`;
